### ISSUE TITLE
Add cache for module file resolution during 'require'

### ISSUE THREAD

**Issue by daiyip** [Microsoft/napajs#37]
## Code
```js
function test() {
    var m = require('some-module');
}
```
## Expected
Only performance hit from require.resolve on first require, afterwards it would be very fast.

## Actual
We only have module cache on canonical path, every `require` will do `require.resolve`, which involves a lot of IO on file lookup.

**Comment by darky** [comment]
Seems it resolved on `0.1.4`.
Communication test 4 workers:

```
time: 19.869ms
time: 19.314ms
time: 19.526ms
time: 19.291ms
time: 0.887ms
time: 0.238ms
time: 0.225ms
time: 0.186ms
time: 0.146ms
time: 0.141ms
time: 0.135ms
time: 0.133ms
time: 0.130ms
time: 0.126ms
time: 0.125ms
```

Seems on first 4 tests you can see, that workers touch IO, and then get from cache.

**Comment by daiyip** [comment]
Are you calling `require` on non-core modules?

There should be 2 level caches:
1) Cache for `require.resolve` - key is (__filename + require-path), value is canonical path.
2) Module cache for loaded modules - key is canonical path.

Now we only have the 2nd cache. The 1st is really expensive, since it needs to recursively lookup files following the node module resolution rule. Well, we might hit system cache for file read (it might be the case if your test is on non-core modules), but it's always good to have an application cache to have consistent performance boost.

**Comment by darky** [comment]
I require **lodash** npm module in tests

**Comment by daiyip** [comment]
Resolved by #128
